docs/concepts/architecture.md

```diff
@@ -0,0 +1,253 @@
+
+```mermaid
+flowchart LR
+    Source[Source Repo] --> Detect{Technology<br/>Detection}
+    Detect -->|Chef| Chef[Chef Agent]
+    Detect -->|Puppet| Puppet[Puppet Agent]
+    Detect -->|Salt| Salt[Salt Agent]
+
+    Chef --> Plan[Module Migration Plan]
+    Puppet --> Plan
+    Salt --> Plan
+
+    style Detect fill:#fff3e0
+    style Plan fill:#e8f5e9
+```
+
+**Chef Agent** (`src/inputs/chef.py`)
+- Uses Tree-sitter for Ruby parsing
+- Integrates Chef Workstation CLI
+- Analyzes recipes, templates, attributes
+- Resolves external cookbook dependencies
+
+**Puppet/Salt Agents**
+- Framework not yet implemented
+- Technology-specific implementation pending
+
+### 4. Export Agents (Migration)
+
+**Migration Agent** (`src/exporters/migrate.py`)
+- Reads migration specifications
+- Generates Ansible playbooks and roles
+- Converts templates (`.erb` → `.j2`)
+- Maps resources to Ansible modules
+
+**Validation Agent** (`src/validate.py`)
+- Runs ansible-lint  and Ansible role check on generated code
+- Auto-retry loop (up to multiple attempts)
+- Compares logic equivalence
+- **Output**: Validation report
+
+### 5. LLM Integration Layer
+
+Supports multiple LLM providers for flexibility and compliance:
+
+| Provider | Use Case | Configuration |
+|----------|----------|---------------|
+| AWS Bedrock | Enterprise, regulated environments | `AWS_BEARER_TOKEN_BEDROCK` |
+| OpenAI API | Development, testing | `OPENAI_API_KEY` |
+| Local Models | Air-gapped networks | `OPENAI_API_BASE` (Ollama/vLLM) |
+
+## Data Flow
+
+```mermaid
+flowchart TB
+    subgraph Input["Input Phase"]
+        Source[Chef Repository]
+        Source --> Init1[1. Init Scan]
+        Init1 --> HLP[migration-plan.md]
+    end
+
+    subgraph Analysis["Analysis Phase"]
+        HLP --> Select[2. Select Module]
+        Select --> Analyze1[Analyze Module]
+        Analyze1 --> Deps[Fetch Dependencies]
+        Deps --> Parse[Tree-sitter Parse]
+        Parse --> Report[Generate Report]
+        Report --> Validate1[Validate Files]
+        Validate1 --> Cleanup[Cleanup Spec]
+        Cleanup --> MMP[migration-plan-module.md]
+    end
+
+    subgraph Migration["Migration Phase"]
+        MMP --> Read[3. Read Plans]
+        HLP --> Read
+        Read --> Choose[Choose Strategy]
+        Choose --> Generate[Generate Ansible]
+        Generate --> Lint{ansible-lint}
+        Lint -->|Pass| Output[Ansible Role]
+        Lint -->|Fail| Fix[Auto-Fix]
+        Fix --> Generate
+        Fix -.->|Max 5| Output
+    end
+
+    subgraph Validation["Validation Phase"]
+        Output --> Compare[4. Compare Logic]
+        Compare --> Final[Validation Report]
+    end
+
+    style Input fill:#e3f2fd
+    style Analysis fill:#e8f5e9
+    style Migration fill:#fff3e0
+    style Validation fill:#f3e5f5
+```
+
+## Tool Integration
+
+Each agent has access to specialized tools:
+
+### Init Agent Tools
+- `FileSearchTool`: Find files by pattern
```

Comment:
Good to have it documented but this will be hard to maintain.

What about sort of:
Each agent has access to specialized tools, which include following:
....

For the up-to-date list, see file path/foo.py